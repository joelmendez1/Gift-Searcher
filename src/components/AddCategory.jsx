import { useState } from 'react';
import PropTypes from "prop-types"

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();

    if(newInputValue.length <= 1) return;

    onNewCategory(newInputValue);
    setInputValue('');
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input 
      type='text' 
      placeholder='Buscar gifs'
      value={inputValue}
      onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
