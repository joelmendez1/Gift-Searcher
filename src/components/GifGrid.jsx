import React, { useEffect, useState } from 'react'
import { getGifs } from '../utils/getGifs'
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category)
      .then(res => setImages(res))
  }, [category])

  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {
          images.map((image) => {
            return <GifItem key={image.id} {...image}/>
          })
        }
      </div>
    </>
  )
}
