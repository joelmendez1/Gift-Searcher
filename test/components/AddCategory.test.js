import { AddCategory } from "../../src/components/AddCategory";
import { fireEvent, render, screen } from "@testing-library/react";


describe('Test in AddCategory component', () => {
  const onNewCategory = () => { };

  test('should match snapshot', () => {
    const { container } = render
      (
        <AddCategory onNewCategory={onNewCategory} />
      )

    expect(container).toMatchSnapshot();
  });

  test('should change input content', () => {
    render
      (
        <AddCategory onNewCategory={onNewCategory} />
      )
    const input = screen.getByRole('textbox');

    fireEvent.input(input, { target: { value: "Saitama" } });

    expect(input.value).toBe("Saitama");
  });
});
