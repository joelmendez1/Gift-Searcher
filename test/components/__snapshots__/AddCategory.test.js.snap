// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory component should match snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
